backend/module/chaos/experimentation/xds/poller.go

```diff
@@ -0,0 +1,311 @@
+package xds
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	gcpRuntimeServiceV3 "github.com/envoyproxy/go-control-plane/envoy/service/runtime/v3"
+	gcpTypes "github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	gcpCacheV3 "github.com/envoyproxy/go-control-plane/pkg/cache/v3"
+	pstruct "github.com/golang/protobuf/ptypes/struct"
+	"github.com/mitchellh/hashstructure/v2"
+	"github.com/uber-go/tally"
+	"go.uber.org/zap"
+
+	experimentation "github.com/lyft/clutch/backend/api/chaos/experimentation/v1"
+	"github.com/lyft/clutch/backend/service/chaos/experimentation/experimentstore"
+)
+
+type Poller struct {
+	ctx           context.Context
+	storer        experimentstore.Storer
+	snapshotCache gcpCacheV3.SnapshotCache
+
+	resourceTtl          time.Duration
+	cacheRefreshInterval time.Duration
+
+	rtdsConfig *RTDSConfig
+	ecdsConfig *ECDSConfig
+
+	rtdsGeneratorsByTypeUrl map[string][]RTDSResourceGenerator
+	ecdsGeneratorsByTypeUrl map[string][]ECDSResourceGenerator
+
+	rtdsResourceGenerationFailureCount        tally.Counter
+	ecdsResourceGenerationFailureCount        tally.Counter
+	ecdsDefaultResourceGenerationFailureCount tally.Counter
+	setCacheSnapshotSuccessCount              tally.Counter
+	setCacheSnapshotFailureCount              tally.Counter
+	activeFaultsGauge                         tally.Gauge
+
+	logger *zap.SugaredLogger
+	ticker *time.Ticker
+}
+
+func (p *Poller) Start() {
+	p.ticker = time.NewTicker(p.cacheRefreshInterval)
+	go func() {
+		for range p.ticker.C {
+			p.logger.Info("Refreshing xDS cache")
+			p.refreshCache()
+		}
+	}()
+}
+
+func (p *Poller) Stop() {
+	p.ticker.Stop()
+}
+
+func (p *Poller) refreshCache() {
```

Comment:
The new implementation of this method tries to follow what the old implementation did - it's just that we use pluggable generators now.